File: app/src/main/java/cg/lastfm/di/AppComponent.java
package cg.lastfm.di;

import android.app.Application;

import javax.inject.Singleton;

import cg.lastfm.ArtistDetailsActivity;
import cg.lastfm.LastFMApp;
import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjectionModule;
import dagger.android.AndroidInjector;

@Singleton
@Component(modules = {AndroidInjectionModule.class, AppModule.class, MainActivityModule.class, ArtistDetailsActivityModule.class})
public interface AppComponent extends AndroidInjector<LastFMApp> {
}

File: app/src/main/java/cg/lastfm/di/AppComponent.kt
package cg.lastfm.di

import cg.lastfm.LastFMApp
import dagger.Component
import dagger.android.AndroidInjectionModule
import dagger.android.AndroidInjector
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AndroidInjectionModule::class, AppModule::class, MainActivityModule::class, ArtistDetailsActivityModule::class))
interface AppComponent : AndroidInjector<LastFMApp>
